Develop

Viacheslav417865 opened the pull request.

Viacheslav417865 committed:
Add and update models, migrations, and views for custom user and movie image upload

Viacheslav417865 committed:
Fix movie image handling and serializer updates

mateacademy-ai-mentor commented on cinema/migrations/0003_alter_ticket_options_alter_movie_image.py:
```diff
@@ -0,0 +1,23 @@
+        ),
+        migrations.AlterField(
+            model_name='movie',
+            name='image',
+            field=models.ImageField(blank=True, null=True, upload_to=cinema.models.movie_image_file_path),
```
Ensure that `cinema.models.movie_image_file_path` is defined in your models module. This function should handle the logic for generating the file path for uploaded images, as required by the task description.

mateacademy-ai-mentor reviewed (changes requested):
Thank you for your hard work on this Pull Request! 😊 However, there are a few critical issues that need to be addressed before we can proceed:

1. **ImageField in Movie Model**: The `Movie` model should include an `ImageField` with a specific upload path format using `slugify` and `uuid`. Please ensure that `cinema.models.movie_image_file_path` is defined and correctly used for this purpose.

2. **Image Upload Endpoint**: Implement the `/upload-image/` endpoint to handle image uploads for the `Movie` model, ensuring images are uploaded only through this endpoint.

3. **Image URL Display**: Ensure the image URL is displayed correctly on the `list` and `detail` pages for movies and movie sessions as specified.

4. **Custom AuthTokenSerializer**: Verify that the `AuthTokenSerializer` is customized to obtain a token using email and password.

5. **User Model Adjustments**: Ensure the `User` model replaces the username field with an email field, and that migrations reflect this change.

Once these issues are resolved, your code will be in great shape! Keep up the great work, and don't hesitate to reach out if you need any help. 🚀

✨ Please rate this review in the Platform's notification center. I will be grateful for your feedback! ✨

🔄 Fix the issues, commit the changes, and then re-request my review using the `Re-request review` button in the Reviewers tab.

Viacheslav417865 commented:
(venv) PS C:\Users\sfink\PycharmProjects\djangoProject\py-images-and-custom-auth-serializer> git status
On branch develop
Changes to be committed:
  (use "git restore --staged <file>..." to unstage)
        modified:   cinema/migrations/0003_alter_ticket_options_alter_movie_image.py
        modified:   cinema/models.py
        modified:   cinema/serializers.py
        modified:   cinema/views.py
        modified:   cinema_service/settings.py
        new file:   media/uploads/movie/sample-movie-0b86a25b-2521-4ccc-bc04-8dd0e44853ff.jpg
        new file:   media/uploads/movie/sample-movie-37aeb294-2b3c-4ff3-8aa9-0daca1e28127.jpg
        new file:   media/uploads/movie/sample-movie-3edd7a92-278f-468e-b8c0-f1155fabb03b.jpg
        new file:   media/uploads/movie/title-6b143345-45e3-419d-a3a2-005bda0ef6d4.jpg
        new file:   media/uploads/movie/title-ff1440bf-a9e8-48cf-9cd0-0e82c4dced6b.jpg
        modified:   requirements.txt
        modified:   user/admin.py

(venv) PS C:\Users\sfink\PycharmProjects\djangoProject\py-images-and-custom-auth-serializer>

Viacheslav417865 committed:
Fix movie image upload view and update related files

mateacademy-ai-mentor commented on cinema/migrations/0002_movie_image.py:
```diff
@@ -0,0 +1,18 @@
+    operations = [
+        migrations.AddField(
+            model_name='movie',
+            name='image',
+            field=models.ImageField(blank=True, null=True, upload_to='uploads/movies/'),
```
The `upload_to` parameter should use a custom function to generate the file path using `slugify` and `uuid`. Please ensure that `cinema.models.movie_image_file_path` is defined and used here to meet the task requirements.

mateacademy-ai-mentor commented on cinema/urls.py:
```diff
@@ -18,6 +17,8 @@
 router.register("orders", OrderViewSet)
 
-urlpatterns = [path("", include(router.urls))]
+urlpatterns = [
+    path("", include(router.urls)),
+]
```
Consider adding a specific endpoint for image uploads to handle image uploads for the `Movie` model, as required by the task description.

mateacademy-ai-mentor commented on user/migrations/0002_customuser.py:
```diff
@@ -0,0 +1,43 @@
+                ('id', models.BigAutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
+                ('password', models.CharField(max_length=128, verbose_name='password')),
+                ('last_login', models.DateTimeField(blank=True, null=True, verbose_name='last login')),
+                ('is_superuser', models.BooleanField(default=False, help_text='Designates that this user has all permissions without explicitly assigning them.', verbose_name='superuser status')),
+                ('username', models.CharField(error_messages={'unique': 'A user with that username already exists.'}, help_text='Required. 150 characters or fewer. Letters, digits and @/./+/-/_ only.', max_length=150, unique=True, validators=[django.contrib.auth.validators.UnicodeUsernameValidator()], verbose_name='username')),
```
If the task requires using only the email field for authentication, consider removing the `username` field from the `CustomUser` model.

mateacademy-ai-mentor reviewed (approved):
Great job on the Pull Request! 🎉 While there are a few areas that need attention, such as ensuring the `ImageField` in the `Movie` model uses the correct custom path format function and defining the `cinema.models.movie_image_file_path`, your overall implementation is solid. The decision to approve is made to help you move forward with your learning journey. Please review the comments to improve your code, and remember to address critical issues in the future to avoid similar problems. Keep up the good work and continue learning from your experiences! 🚀

✨ Please rate this review in the Platform's notification center. I will be grateful for your feedback! ✨
